code/game/objects/items/devices/radio/headset.dm

```diff
@@ -417,8 +417,10 @@
 	var/obj/item/card/id/id_card = wearer.wear_id?.GetID()
 	if(!istype(id_card))
 		return
-	if(!(id_card.rank in list(JOB_SO, JOB_XO, JOB_SQUAD_LEADER)))
-		to_chat(wearer, SPAN_WARNING("Only Staff Officers, Executive Officers and Squad Leaders are permitted to give medal recommendations!"))
+	var/paygrade = get_paygrade_id_by_name(wearer.get_paygrade(FALSE))
+	var/last_number = text2num(copytext(paygrade, length(paygrade), length(paygrade)+1))
+	if(!(findtext(paygrade, "O") || last_number >= 4))
+		to_chat(wearer, SPAN_WARNING("Only officers or NCO's (ME4+) can recommend medals!"))
```

Comment:
String manipulation is not the way to go about this. Either just explicitly list out the ranks that are allowed, or consider a more appropriate value e.g. `officer_grade` on the `/datum/paygrade`

Reply:
You’ll be having a really long list, and officer_grade doesn’t account for NCOs.
There’s nothing wrong with string manipulation.